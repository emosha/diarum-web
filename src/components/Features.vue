<template>
  <div class="features" id="features">
    <h2 class="features__heading">Features</h2>
    <hr class="features__line">
    <div class="features__cards">
      <FeatureCard text="Record your day" icon="pen" />
      <FeatureCard text="Add your mood" icon="smile" />
      <FeatureCard text="Check your todos" icon="check"  />
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';
import FeatureCard from '@/components/shared/FeatureCard.vue';

@Component({
  components: {
    FeatureCard,
  },
})

export default class Features extends Vue {}
</script>

<style scoped lang="scss">
  @import '@/assets/styles/mixins.scss';

  .features {
    background: url('../assets/img/diarum-features.svg') no-repeat center center;
    background-size: cover;
    width: 100%;
    padding-top: 70px;

    &__heading {
      color: #5d59cb;
      font-weight: 600;
      text-transform: uppercase;
      letter-spacing: 2.5px;
      font-size: 1.8em;
      margin-bottom: 15px;
    }

    &__line {
      width: 8%;
      border-top: 3px solid #5d59cb;
    }

    &__cards {
      display: flex;
      justify-content: space-evenly;
      align-items: center;
      padding-top: 130px;

      .feature-card {
        display: flex;
        flex-direction: column;
        justify-content: space-around;
        align-items: center;
        transition: .3s margin-top ease;

        &:hover {
          margin-top: -20px;
        }
      }
    }
  }

  @media (max-width: 1199.98px) {
    .features {
      padding-top: 40px;

      &__cards {
        display: flex;
        flex-direction: column;
        justify-content: space-between;
        align-items: center;
        padding-top: 100px;

        .feature-card {
          margin-bottom: 30px;
        }
      }
    }
  }

  @include media-breakpoint-down(sm) {
    .features {
      padding-top: 0;
    }
  }
</style>
