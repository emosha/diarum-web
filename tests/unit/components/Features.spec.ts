import { shallowMount } from '@vue/test-utils';
import Features from '@/components/Features.vue';

describe('Features.vue', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(Features);
    expect(wrapper).toMatchSnapshot();
  });
});
