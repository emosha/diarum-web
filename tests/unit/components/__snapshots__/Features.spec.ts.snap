// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features.vue renders correctly 1`] = `
<div id="features" class="features">
  <h2 class="features__heading">Features</h2>
  <hr class="features__line">
  <div class="features__cards">
    <featurecard-stub text="Record your day" icon="pen"></featurecard-stub>
    <featurecard-stub text="Add your mood" icon="smile"></featurecard-stub>
    <featurecard-stub text="Check your todos" icon="check"></featurecard-stub>
  </div>
</div>
`;
